### ISSUE TITLE
Add support for images from Commons

### ISSUE THREAD

**Issue by DerDings** [MarcusWolschon/osmeditor4android#1247]
Since showing images from the local device storage and from Mapillary is already supported, adding a layer with images from Wikimedia Commons seems to be the logical next step.

The POI-focused lineup of the Commons database is the perfect supplement for the street-focused character of Mapillary. 

Features should be the same as for the other photo resources:
- show images in the place of the camera's location
- show camera's compass angle if available
- layer is not enabled by default

**Comment by simonpoole** [comment]
Is there a corresponding API for Wikimedia Commons that allows querying per bounding box and will return image meta data with location?

**Comment by DerDings** [comment]
> Is there a corresponding API for Wikimedia Commons that allows querying per bounding box and will return image meta data with location?

Since [this](https://wikimap.toolforge.org/) is able to show the place and direction of images, I hope there is. But I am not a programmer so I don't know about the details. See [this location](https://www.openstreetmap.org/?mlat=51.422675&mlon=7.258548&zoom=15#map=15/51.4227/7.2585) for examples.

**Comment by simonpoole** [comment]
The wikimap tool doesn't seem to use a public api that I could find, it seems (source code doesn't seem to be available) to run a bot to create a private database that is then used to enable the queries. Naturally that means that wikimap could conceivably offer such an API call itself, but currently it doesn't seem to do so.  

Please feel free to contact the wikimap maintainer and point him to this discussion.

**Comment by DerDings** [comment]
> Please feel free to contact the wikimap maintainer and point him to this discussion.

The maintainer of the wikimap (thanks, DB111) pointed me to the [GeoSearchAPI](https://commons.wikimedia.org/w/api.php?action=query&list=geosearch&gsbbox=37.8%7C-122.3%7C37.7%7C-122.4&gsnamespace=6). Is this helping?

**Comment by Discostu36** [comment]
Most images in Commons don't have a licence that is compatible for using it for deriving map data from it. So I don't know if this a helpful feature.

**Comment by simonpoole** [comment]
> Most images in Commons don't have a licence that is compatible for using it for deriving map data from it. So I don't know if this a helpful feature. 

Though at least in princple it should be possible to get a machine readable licence identificator from the API.

**Comment by Discostu36** [comment]
Yes, with a query like this you can get the licence, you can even filter for attribution requirement

https://commons.wikimedia.org/w/api.php?action=query&format=json&prop=imageinfo&titles=File:Berliner_Tor_5_(Hamburg-St._Georg).ajb.jpg&iiprop=user|url|extmetadata|metadata&iilimit=1
